### PR TITLE
Iterated bug

### PR DESCRIPTION
**Pull request by joakim-hove**
Currently libres/ert has one implementation of an *iterated* smoother - the RML module. This module is quite complex, was from the outset not very well implemented and has had many problems. Partly due to this the gui model class to actually manage/run an iterated smoother has not seen much attention either.

As part of @geirev work: https://github.com/Statoil/libres/pull/422 a significantly improved iterated implementation is on the way, and as preparation for that - this PR fixes a bug in the gui model class for running iterated smoothers.

### CODE REVIEW

File: python/python/ert_gui/simulation/models/iterated_ensemble_smoother.py [hunk 1]
```diff
@@ -76,32 +76,35 @@ def runSimulations(self, arguments):
         analysis_iter_config = analysis_config.getAnalysisIterConfig()
         num_retries_per_iteration = analysis_iter_config.getNumRetries()
         num_tries = 0
-        current_iteration = 1
 
-        while current_iteration <= getNumberOfIterations() and num_tries < num_retries_per_iteration:
+        while True:
```
Comment (markusdregi) [inline python/python/ert_gui/simulation/models/iterated_ensemble_smoother.py]: Would not the old loop condition with `current_iteration <= getNumberOfIterations()` changed to `current_iteration < getNumberOfIterations()` yield same behaviour? If so, I think that would be much easier to read...

File: python/python/ert_gui/simulation/models/iterated_ensemble_smoother.py [hunk 2]
```diff
@@ -76,32 +76,35 @@ def runSimulations(self, arguments):
...
             pre_analysis_iter_num = analysis_module.getInt("ITER")
             self.analyzeStep( run_context )
-            post_analysis_iter_num = analysis_module.getInt("ITER")
+            current_iter = analysis_module.getInt("ITER")
 
             analysis_success = False
-            if  post_analysis_iter_num > pre_analysis_iter_num:
+            if current_iter > pre_analysis_iter_num:
```
Comment (markusdregi) [inline python/python/ert_gui/simulation/models/iterated_ensemble_smoother.py]: Could not these lines be replaced by;
 ```python
analysis_success = current_iter > pre_analysis_iter_num
if analysis_success:
    .....
```
?

File: test-data/local/poly_example/poly.ert [hunk 3]
```diff
@@ -16,3 +16,6 @@ GEN_DATA POLY_RES RESULT_FILE:poly_%d.out REPORT_STEPS:0 INPUT_FORMAT:ASCII
 
 INSTALL_JOB poly_eval POLY_EVAL
 SIMULATION_JOB poly_eval
+-- Load the RML module, this does not imply that it is actually used. That will
```
Comment (markusdregi) [inline test-data/local/poly_example/poly.ert]: Could you please move this to a separate config file (`poly_rml_enkf.ert` or similar)? I really like the simplicity of the `poly.ert` config file and I'm planning to use it in early training of new developers...

File: python/python/ert_gui/simulation/models/iterated_ensemble_smoother.py (hunk 2 above)
Comment (joakim-hove) [reply python/python/ert_gui/simulation/models/iterated_ensemble_smoother.py]: Yes - done.

File: python/python/ert_gui/simulation/models/iterated_ensemble_smoother.py (hunk 1 above)
Comment (joakim-hove) [reply python/python/ert_gui/simulation/models/iterated_ensemble_smoother.py]: I'll look into this; personally I kind of like the:
```
while True:

    if cond1: 
         break

    if cond2:
         break
```
pattern; but the is in the beholder. Anyway - will look into your suggestion.

File: test-data/local/poly_example/poly.ert (hunk 3 above)
Comment (joakim-hove) [reply test-data/local/poly_example/poly.ert]: Done

**Pull request merged**